internal/services/applications/parse/application.go

```diff
@@ -68,3 +68,13 @@ func (id ApplicationId) Segments() []resourceids.Segment {
 func (id ApplicationId) String() string {
 	return fmt.Sprintf("Application (Object ID: %q)", id.ApplicationId)
 }
+
+func (id ApplicationId) FromParseResult(input resourceids.ParseResult) error {
```

Comment:
This'll need to be a pointer:

```suggestion
func (id *ApplicationId) FromParseResult(input resourceids.ParseResult) error {
```